Reflected that HTTP was updated again in 2022 with RFC 9110

Ashish-CodeJourney opened the pull request:
### Description

<!-- ✍️ Summarize your changes in one or two sentences -->
Updated the documentation to reflect that HTTP was updated in 2022 with RFC 9110.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->
The previous documentation did not include the most recent update to HTTP, which occurred in 2022 with RFC 9110. This update is important for providing readers with the most current information regarding the HTTP protocol.

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->
- [RFC 9110 - HTTP Semantics](https://www.rfc-editor.org/rfc/rfc9110.html)
- [HTTP/1.1 Overview](https://httpwg.org/specs/rfc9110.html)

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->
Fixes #34374

Ashish-CodeJourney committed:
Relected that HTTP was updated again in 2022 with `RFC 9110` #34376

Ashish-CodeJourney committed:
minor changes

Ashish-CodeJourney committed:
Update files/en-us/learn/javascript/objects/basics/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Ashish-CodeJourney committed:
revert old changes

Ashish-CodeJourney committed:
revert old changes

JannisBush commented on files/en-us/web/http/basics_of_http/evolution_of_http/index.md:
```diff
@@ -142,7 +142,7 @@ HTTP/1.1 was first published as {{rfc(2068)}} in January 1997.
 ## More than 15 years of extensions
 
-The extensibility of HTTP made it easy to create new headers and methods. Even though the HTTP/1.1 protocol was refined over two revisions, {{RFC("2616")}} published in June 1999 and {{RFC("7230")}}-{{RFC("7235")}} published in June 2014 before the release of HTTP/2, it was extremely stable for more than 15 years.
+The extensibility of HTTP made it easy to create new headers and methods. Even though the HTTP/1.1 protocol was refined over two revisions, {{RFC("2616")}} published in June 1999 and {{RFC("7230")}}-{{RFC("7235")}} published in June 2014 before the release of HTTP/2, it was extremely stable for more than 15 years. HTTP/1.1 was updated again in 2022 with {{RFC("9110")}}.
```
Not only HTTP/1.1 was updated, but all of HTTP which is now split into semantics (9110), caching (9111) applying to all HTTP versions and HTTP/1.1 (9112), HTTP/2 (9113) and HTTP/3 (9114).

In addition, the spec finally achieved the status of internet standard (STD 97) whereas before it was always a proposed/draft standard.

Ashish-CodeJourney replied:
Incorporated ! Please review

JannisBush replied:
They year (2022) got lost in the update.

Ashish-CodeJourney replied:
done, thanks for review :)

Ashish-CodeJourney committed:
Update HTTP/1.1 documentation with latest RFCs and status

- Added reference to RFC 9110, 9111, 9112, 9113, and 9114 which outline the latest revisions to HTTP semantics, caching, HTTP/1.1, HTTP/2, and HTTP/3 respectively.
- Mentioned the transition of the HTTP specification to Internet Standard (STD 97).
- Ensured clarity and accuracy regarding the historical context and updates of HTTP protocols.

Ashish-CodeJourney committed:
Suggestions incorporated

bsmth commented on files/en-us/web/http/basics_of_http/evolution_of_http/index.md:
```diff
@@ -142,7 +142,7 @@ HTTP/1.1 was first published as {{rfc(2068)}} in January 1997.
 
 ## More than 15 years of extensions
```
The only last thing to look at is maybe changing the section title here:

```suggestion
## More than two decades of development
```

Ashish-CodeJourney committed:
suggestions incorporated

Co-authored-by: Brian Thomas Smith <brian@smith.berlin>

bsmth reviewed (approved):
Thanks for the improvements, going to merge shortly!

pransh15 commented:
Well done, @Ashish-CodeJourney! 👏
